frontend/ui/widget/touchmenu.lua

```diff
@@ -992,260 +989,266 @@ function TouchMenu:onBack()
+                if with_animation and (pages_diff > 1 or pages_diff < -1) then
+                    -- Change pages quicker if more than one needed, but slow on the last one
+                    next_delay = 0.5
+                end
             end
-        end
-
-        local function open_next_menu()
-            if animation_time_s == 0.0 then
-                UIManager:close(self.trap_widget)
-                self.trap_widget = nil
-                open_final_menu()
-                -- end of the animation here!
-                return
+        elseif step == STEPS.MENU_ITEM_HIGHLIGHT then
+            if with_animation or #parts == 0 then
+                -- Even if no animation, highlight the final item (and don't unhighlight it)
+                local item_visible_index = (item_nb - 1) % self.perpage + 1
+                local item_widget
+                for i, w in ipairs(self.item_group) do
+                    if w.item_visible_index == item_visible_index then
+                        item_widget = w
+                        break
+                    end
+                end
+                if item_widget then
+                    highlightWidget(item_widget)
+                end
             end
-
-            local x = table.remove(items_to_show, 1)
-            local next_menu_num, next_menu_item = x and x[1], x and x[2] -- might be nil
-
-            if next_menu_item then
-                highlightWidget(self:_get_widget(tab_nb, next_menu_num))
-                UIManager:scheduleIn(animation_time_s, function()
-                    self:onMenuSelect(next_menu_item)
-                    UIManager:nextTick(open_next_menu)
-                end)
+            if #parts == 0 then
+                step = STEPS.DONE
             else
-                -- no items left, but maybe on another page
-                local page_nb = math.floor(item_nb / perpage) + 1
-                UIManager:nextTick(open_next_page, page_nb, page_nb > 1)
+                step = STEPS.MENU_ITEM_ENTER
+            end
+        elseif step == STEPS.MENU_ITEM_ENTER then
+            self:onMenuSelect(self.item_table[item_nb])
+            item_nb = table.remove(parts)
+            step = STEPS.TARGET_PAGE_OR_HIGHLIGHT_NEXT_PREV
+        else -- STEPS.DONE
+            if trap_widget then
+                UIManager:close(trap_widget)
+                trap_widget = nil
             end
+            return
+        end
+        if next_delay >= 0 then
+            walkStep_scheduled = true
+            UIManager:scheduleIn(next_delay, walkStep)
+        else
+            walkStep()
         end
-
-        -- Animate
-        UIManager:nextTick(open_next_menu)
     end
+
+    -- We use an invisible TrapWidget when no animation, so we can
+    -- cancel the delayed final highlight
+    local TrapWidget = require("ui/widget/trapwidget")
+    trap_widget = TrapWidget:new{
+        text = with_animation and "Walking you there…" or nil,
```

Comment:
That's actually invisible, hence the lack of i18n, right?

Reply:
No, it is visible, I just forgot the `_()`. Will add them.